manifests/generate_archive.pp

```diff
@@ -0,0 +1,46 @@
+# Prepare the certificates for the node from the parent node
+#
+# === Parameters:
+#
+# $server_fqdn::                   FQDN of the Foreman app
+#
+# $server_cname::                  Additional names of the foreman app
+#
+# $certs_tar::                     Path to tar file with certs to generate
+#
+# $foreman_application::           Create certificate bundle for a Foreman node
+#
+# $foreman_proxy::                 Create certificate bundle for a Foreman Proxy node
+#
+class certs::generate_archive (
+  String $server_fqdn          = undef,
+  Array[String] $server_cname  = $::certs::params::cname,
+  String[1] $certs_tar         = $::certs::params::certs_tar,
+  Boolean $foreman_application = false,
```

Comment:
This is the general direction I'm going for but I'm not sure this is the granularity I'm aiming for. My goal is to split the node with candlepin and pulp into two separate nodes. Ideally qpid could run on a third. That means we'd bootstrap with candlepin as a node with the CA, then generate the certs for each additional node with the role they have. Do you think this makes sense?

Reply:
Yes, and I agree its what we should aim for, well my end goal is being able to scale each component independently of each other.

Initially, I think we should try to make small(er) changes and iterate on them. Fewer chances of breakages and easier to review, we can extend the scope in other PRs later.

Reply:
I've marked it as WIP for now so we don't accidentally merge it. I agree we should iterate on this but since it's a breaking change I'd like to keep it open until we're really happy with it.